libs/providers/langchain-anthropic/src/utils/message_inputs.ts

```
@@ -271,6 +272,15 @@ function* _formatContentBlocks(
         ...(cacheControl ? { cache_control: cacheControl } : {}),
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
       } as any;
+    } else if (contentPart.type === "container_upload") {
+      yield {
+        ...contentPart,
+        ...(cacheControl ? { cache_control: cacheControl } : {}),
+      } as AnthropicContainerUploadBlockParam;
+    } else {
+      console.warn(
+        `Skipping unsupported content block of type "${contentPart.type}".`
+      );
```

One big gotcha here for me was that we silently drop unknown blocks. I turned it into a warning, but my instinct would be to warn and then just pass them through untouched. Any thoughts?

---

I would throw here because it means we not fully support Anthropic's API which causes such unknown errors.

---

One consideration is that message .content looks subtly different between providers, which means if I used openai generated messages in an anthropic call and had the content fall through we'd always get 400 errors. I also don't think it's great DX to flood the console with these warning messages if we know that some messages aren't compatible with anthropic to boot.

Generally when serializing provider requests, we take the approach of "best effort" parsing where we serialize messages for a request to the best of our ability (typically by having explicit conversion for messages that come out of the same provider + some other well known formats), but we don't explicitly fail since we can't interpret every possible message format (n^n complexity)

this is part of the motivation for standard content in v1!